### PR TITLE
userdata=None

### PR DESCRIPTION
**Pull request by MatthijsBurgh**

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
Sorry for the one wrong commit message, too quick with ctrl+c & ctrl+v :p

File: robot_smach_states/examples/random_nav2.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def __init__(self, robot):
             @smach.cb_interface(outcomes=['target_determined', 'no_targets_available'],
                                 input_keys=[],
                                 output_keys=[])
-            def determine_target(userdata, designator):
+            def determine_target(userdata=None, designator):
```
Comment (LoyVanBeek) [inline robot_smach_states/examples/random_nav2.py]: Python does not allow keyword arguments before other arguments AFAIK

**Comment by Rayman** [comment]
Can you explain why you made this PR?

**Comment by MatthijsBurgh** [comment]
So you can use smach.states outside a state machine

**Comment by MaxBaeten** [comment]
You can just do:
state.execute({})

**Comment by MatthijsBurgh** [comment]
@LoyVanBeek, what needs to be changed? We discussed it last Tuesday, but not thoroughly

**Comment by LoyVanBeek** [comment]
There are 3 options:

- Do not do the ```userdata=None```, but leave just ```userdata```. This requires the user to have a knowledge of python
- also set ```designator=None``` but this might cause issues because None makes no sense. Perhaps a different default argument but I can't think of a sensible value
- Swap the order of arguments, but then you also have to check all callers to also change this order.

**Pull request merged**